### PR TITLE
Add Mozilla Addons support

### PR DESCRIPTION
**Pull request by qzb**
Version, downloads, rating and daily users badges for mozilla addons.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -5183,6 +5183,68 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Mozilla addons integration
+camp.route(/^\/amo\/(v|d|rating|users)\/(.*)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var xml2js = require('xml2js');
```
Comment (espadrine) [inline server.js]: Could you put this at the start of the file, please?

**Comment by RedSparr0w** [comment]
Nice work,
someone please merge this

**Comment by qzb** [comment]
@espadrine I understand that there is a plenty open pull requests and you probably don't have any time for cleaning them up, but do you plan to merge this one someday?

**Comment by RedSparr0w** [comment]
@espadrine Could you please have a look at merging this?

**Comment by RedSparr0w** [comment]
if you add `?format=json` to the end of the api request that would remove the xml2js dependency

also in try.html on [Line 394](https://github.com/badges/shields/pull/722/files#diff-20b51e3176fb8c62d5ce21aee43e6cd1R394) remove the double `::`

**Comment by qzb** [comment]
Unfortunately JSON response doesn't contain all fields of XML response.  For example, there is no *rating* and *daily users*.

**Comment by RedSparr0w** [comment]
Strange, didn't notice that.
I wonder why they don't include that information.

File: try.html [hunk 2]
```diff
@@ -384,13 +384,16 @@ <h3 id="downloads"> Downloads </h3>
...
+  <td><img src='/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg' alt=''/></td>
+  <td><code>https://img.shields.io/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg</code></td>
   </tr>
   <tr><th data-keywords='website' data-doc='websiteDoc'> Website: </th>
   <td><img src='/website-up-down-green-red/http/shields.io.svg' alt=''/></td>
   <td><code>https://img.shields.io/website-up-down-green-red/http/shields.io.svg</code></td>
   </tr>
+  <tr><th data-keywords='amo firefox mozilla'> Mozilla Addons: </th>
+  <td><img src='/amo/d/dustman.svg' alt=''/></td>
+  <td><code>https://img.shields.io/amo/d/dustman.svg</code></td>
```
Comment (espadrine) [inline try.html]: Could you add the missing `</tr>` please?

File: try.html [hunk 3]
```diff
@@ -384,13 +384,16 @@ <h3 id="downloads"> Downloads </h3>
...
-    <td><img src='/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg' alt=''/></td>
-    <td><code>https://img.shields.io/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg</code></td>
+  <td><img src='/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg' alt=''/></td>
+  <td><code>https://img.shields.io/chrome-web-store/d/nimelepbpejjlbmoobocpfnjhihnpked.svg</code></td>
   </tr>
   <tr><th data-keywords='website' data-doc='websiteDoc'> Website: </th>
   <td><img src='/website-up-down-green-red/http/shields.io.svg' alt=''/></td>
   <td><code>https://img.shields.io/website-up-down-green-red/http/shields.io.svg</code></td>
   </tr>
+  <tr><th data-keywords='amo firefox mozilla'> Mozilla Addons: </th>
```
Comment (espadrine) [inline try.html]: You can remove the `mozilla` keyword, since it is already in the th.

File: package.json [hunk 4]
```diff
@@ -26,7 +26,8 @@
     "semver": "~4.3.3",
     "bower": "~1.4.1",
     "chrome-web-store-item-property": "^1.1.2",
-    "json-autosave": "~1.1.1"
+    "json-autosave": "~1.1.1",
+    "xml2js": "^0.4.16"
```
Comment (espadrine) [inline package.json]: Please use `~`.

**Comment by espadrine** [comment]
I squashed it, changed it and merged it in 2cb82a4ab32974052e8e3a0eb3b72f8e728c2c9c.

I added a few changes in 7403b61e253776c75f18592110f2bf69bd500e01.

**Comment by espadrine** [comment]
Thanks for your contribution!

This is now live.

**Comment by qzb** [comment]
Thanks :)

**Pull request closed without merging**